src/api/messages/updateMessageFlagsForNarrow.js

```diff
@@ -0,0 +1,32 @@
+// @flow strict-local
+import type { ApiNarrow, UserMessageFlag } from '../modelTypes';
+import type { ApiResponseSuccess, Auth } from '../transportTypes';
+import { apiPost } from '../apiFetch';
+
+// TODO(server-3.0): Move `Anchor` to modelTypes and share with getMessages.
+type Anchor = number | 'newest' | 'oldest' | 'first_unread';
+
+type ApiResponseUpdateMessageFlagsForNarrow = {|
+  ...$Exact<ApiResponseSuccess>,
+  affected_count: number,
+  first_affected_id: number | null,
+  last_affected_id: number | null,
+  found_oldest: boolean,
+  found_newest: boolean,
+|};
+
+/** https://zulip.com/api/update-message-flags-for-narrow */
+export default function updateMessageFlagsForNarrow(
+  auth: Auth,
+  args: {|
+    narrow: ApiNarrow,
+    anchor: Anchor,
+    include_anchor?: boolean,
+    num_before: number,
+    num_after: number,
+    op: 'add' | 'remove',
+    flag: UserMessageFlag,
+  |},
+): Promise<ApiResponseUpdateMessageFlagsForNarrow> {
+  return apiPost(auth, '/messages/flags/narrow', { ...args, narrow: JSON.stringify(args.narrow) });
+}
```

Comment:
At a dynamic level, such a check isn't really necessary because on an older server, the request will promptly fail because the endpoint doesn't exist -- which isn't much different from having an assertion that fails.

In `api.updateUserSettings`:
```js
/**
 * https://zulip.com/api/update-settings
 *
 * NOTE: Only supports servers at feature level 89+ because UserSettings
 * assumes 89+; see there.
 */
// TODO(server-5.0): Simplify FL 89+ requirement
export default (/* … */): Promise<ApiResponseSuccess> => {
  invariant(zulipFeatureLevel >= 89, 'api.updateUserSettings not supported with FL <89');
```
the check was more necessary because the endpoint itself is older, so that it's more complicated to think about what happens if the server is older -- the endpoint exists and just behaves differently. So the assertion potentially saves us from some more complicated and puzzling behavior we'd get by making the actual request.

Because the assertion could never fire on up-to-date servers, which are what we normally develop against, it also doesn't help find bugs at development time.

There would be value in having the version requirement documented in the jsdoc. In principle it's all there in the linked API docs, though, because (again unlike `api.updateUserSettings`) the version requirement is just restating the "this endpoint was added in …" paragraph from the API docs. So I'm inclined to stick to letting the API docs speak for themselves -- in general when adding or reviewing a new call site for an API endpoint, one is going to need to consult the API docs in any case.

Reply:
Got it, makes sense!